assets/javascripts/birdeck.js

```diff
@@ -17,32 +22,34 @@ function renderPost(post) {
     + "</p>"
     + "<button id='delete-post' name='button-fetch' class='btn btn-default btn-xs'>Delete</button>"
     + "</div>"
-    )
+  )
+}
+
+function renderPosts(posts) {
+  return posts.map(renderPost)
+}
+
+function addPostToPage(post) {
+  $latestPosts.append(post);
+}
+
+function addPostsToPage(posts) {
+  $latestPosts.html(posts)
+}
+
+function handleError(xhr) {
+  console.log(xhr.responseText)
 }
 
 function fetchPostsButton() {
-  $("#button-fetch-posts").on("click", function(){
-    fetchPosts()
-    })
+  $("#button-fetch-posts").on("click", fetchPosts)
 }
 
 function fetchPosts() {
-  var newestItemID = parseInt($(".post").last().attr("data-id"))
-
-  $.ajax({
-    type:    "GET",
-    url:     "https://turing-birdie.herokuapp.com/api/v1/posts.json",
-    success: function(posts) {
-      $.each(posts, function(index, post) {
-        if (isNaN(newestItemID) || post.id > newestItemID) {
-          renderPost(post)
-        }
-      })
-    },
-    error: function(xhr) {
-      console.log(xhr.responseText)
-    }
-  })
+  $.getJSON("https://turing-birdie.herokuapp.com/api/v1/posts.json")
```

Comment:
Yea, we teach them those concepts as building blocks and then we—soon after—show them the normal way. I don't think that we shouldn't mention `$.ajax` but I _do think_ that we should quickly follow up with the generally accepted way of doing it.

I'm find with it being a bridge to get somewhere but the idea that the other end of that bridge doesn't come until weeks later is kind of silly.

There are two issues here:

- `$.ajax` vs. `$.get`, `$.post`, and `$.getJSON`
` Callbacks vs. promises

I don't care about the latter, but the former isn't working as we've seen in Module 4. It causes confusion that could easily be rectified by at least exposing them to the entire landscape when they learn it.